mantidproject/mantid
# Ensure legend stays on top of horizontal and vertical markers

**Description of work.**
This PR ensures that the legend appears above horizontal and vertical markers. Ideally, this would be achieved through setting the z-order, however the markers are specified with `animated=true` which means they are not drawn when `.draw()` is called. Instead, to draw them we must explicitly call `draw_artist` which trashes our z-order and places the marker at the top of the image. Therefore, to keep the legend on top we must redraw the legend patch after redrawing the marker. 


**To test:**

1. Load any data
2. Plot a spectrum and open the fit browser
3. Right click and add a peak
4. All the markers should be behind the legend (test by moving the legend)



Fixes #26553 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
